product/service.go

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package product
+
+type ServiceKey string
+
+const (
+	ChannelKey       ServiceKey = "channel"
+	ConfigKey        ServiceKey = "config"
+	LicenseKey       ServiceKey = "license"
+	FilestoreKey     ServiceKey = "filestore"
+	FileInfoStoreKey ServiceKey = "fileinfostore"
+	ClusterKey       ServiceKey = "cluster"
+	CloudKey         ServiceKey = "cloud"
+	PostKey          ServiceKey = "post"
+	TeamKey          ServiceKey = "team"
+	UserKey          ServiceKey = "user"
+	PermissionsKey   ServiceKey = "permissions"
+	RouterKey        ServiceKey = "router"
+	BotKey           ServiceKey = "bot"
+	LogKey           ServiceKey = "log"
+	HooksKey         ServiceKey = "hooks"
+	KVStoreKey       ServiceKey = "kvstore"
+	StoreKey         ServiceKey = "storekey"
+	SystemKey        ServiceKey = "systemkey"
+	PreferencesKey   ServiceKey = "preferenceskey"
+
+	// The following are not services, but are used for the
+	// other type of entities that might need to be passed to the
+	// initializers.
+
+	// ServerKey is used to pass the app.Server to the initializers.
+	ServerKey ServiceKey = "server"
```

Comment:
I like that server is exposed via service map entry versus passed to the initializers.  However, I don't think it should be exposed at all.  It seems like a short-cut to expose stuff that should have a service interface and key here.